eventtracking/backends/routing.py

```diff
@@ -0,0 +1,131 @@
+       backend will block other backends until it is done persisting the event. Note that you can register another
+       `RoutingBackend` as a backend of a `RoutingBackend`, allowing for arbitrary processing trees.
+
+    `backends` is a collection that supports iteration over it's items using `iteritems()`. The keys are expected to be
+        sortable and the values are expected to expose a `send(event)` method that will be called for each event.
+    `processors` is an iterable of callables.
+
+    Raises a `ValueError` if any of the provided backends do not have a callable "send" attribute or any of the
+        processors are not callable.
+    """
+
+    def __init__(self, backends=None, processors=None):
+        self.backends = OrderedDict()
+        self.processors = []
+
+        if backends is not None:
+            for name in sorted(backends.keys()):
+                self.register_backend(name, backends[name])
+
+        if processors is not None:
+            for processor in processors:
+                self.register_processor(processor)
+
+    def register_backend(self, name, backend):
+        """
+        Register a new backend that will be called for each processed event
+
+        Note that backends are called in the order that they are registered.
+        """
+        if not hasattr(backend, 'send') or not callable(backend.send):
+            raise ValueError('Backend %s does not have a callable "send" method.' % backend.__class__.__name__)
+        else:
+            self.backends[name] = backend
+
+    def register_processor(self, processor):
+        """
+        Register a new processor.
+
+        Note that processors are called in the order that they are registered.
+        """
+        if not callable(processor):
+            raise ValueError('Processor %s is not callable.' % processor.__class__.__name__)
+        else:
+            self.processors.append(processor)
+
+    def send(self, event):
+        """
+        Process the event using all registered processors and send it to all registered backends.
+
+        Logs and swallows all `Exception`.
+        """
+        try:
+            processed_event = self.process_event(event)
+        except EventEmissionExit:
+            return
+        else:
+            self.send_to_backends(processed_event)
+
+    def process_event(self, event):
+        """
+
+        Executes all event processors on the event in order.
+
+        `event` is a nested dictionary that represents the event.
+
+        Logs and swallows all `Exception` except `EventEmissionExit` which is re-raised if it is raised by a processor.
+
+        Returns the modified event.
+        """
+
+        if len(self.processors) == 0:
+            return event
+
+        processed_event = event
+
+        for processor in self.processors:
+            try:
+                modified_event = processor(processed_event)
+                if modified_event is not None:
```

Comment:
That is all correct. Additionally, I like the explictness of raising the exception vs. returning None. It would be easy to write a processor that didn't return from some branch and that would result in the event not being propagated (oops!)